### PR TITLE
Rebase

### PR DESCRIPTION
**Pull request by EnnioSileno**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -32,6 +46,44 @@
...
 
   <dependencies>
+    <dependency>
+      <groupId>net.adoptopenjdk.api</groupId>
+      <artifactId>adoptopenjdk-api-v3-models</artifactId>
+      <version>3.0.0-SNAPSHOT</version>
+    </dependency>
+    <dependency>
+      <groupId>org.jetbrains.kotlin</groupId>
+      <artifactId>kotlin-reflect</artifactId>
```
Comment (hendrikebbers) [inline pom.xml:56]: Why do we depend on Kolin?

File: pom.xml [hunk 2]
```diff
@@ -32,6 +46,44 @@
...
+      <version>2.12.0</version>
+    </dependency>
+    <dependency>
+      <groupId>net.adoptopenjdk</groupId>
+      <artifactId>net.adoptopenjdk.v3.api</artifactId>
+      <version>0.4.0</version>
+    </dependency>
+    <dependency>
+      <groupId>net.adoptopenjdk.api</groupId>
+      <artifactId>adoptopenjdk-api-v3-models</artifactId>
```
Comment (hendrikebbers) [inline pom.xml:76]: This PR adds 2 dependencies to `adoptopenjdk-api-v3-models`. I assume one can be removed?

File: src/main/java/net/adoptium/IndexResource.java [hunk 3]
```diff
@@ -1,25 +1,52 @@
 package net.adoptium;
 
+import io.quarkus.qute.Template;
+import io.quarkus.qute.TemplateInstance;
+import net.adoptium.api.ApiService;
+import net.adoptopenjdk.api.v3.models.*;
```
Comment (hendrikebbers) [inline src/main/java/net/adoptium/IndexResource.java:6]: Let's try to not use * imports

File: src/main/java/net/adoptium/IndexResource.java [hunk 4]
```diff
@@ -1,25 +1,52 @@
...
+                }
+            }
+        }
+        return response;
+    }
+
     @GET
     @Produces(MediaType.TEXT_HTML)
     public TemplateInstance get(@QueryParam("name") String name) {
+        System.out.println("Binary: " + getUserDownload(OperatingSystem.windows, Architecture.x64));
```
Comment (hendrikebbers) [inline src/main/java/net/adoptium/IndexResource.java:49]: Please do not commit System.out ;)

File: src/main/java/net/adoptium/IndexResource.java [hunk 5]
```diff
@@ -1,25 +1,52 @@
...
+
+    private Binary getUserDownload(OperatingSystem os, Architecture arch) {
+        List<BinaryAssetView> availableReleaseList = api.getAvailableReleases(RECOMMENDED_JAVA_VERSION, JvmImpl.hotspot);
+        Binary response = null;
+
+        for (BinaryAssetView release : availableReleaseList) {
+            if (release.getBinary().getOs() == os && release.getBinary().getArchitecture() == arch) {
+                response = release.getBinary();
+                if (release.getBinary().getInstaller() != null) {
+                    break;
```
Comment (hendrikebbers) [inline src/main/java/net/adoptium/IndexResource.java:39]: We can do the return here. By doing the `response` variable can be defined as static.

File: src/main/java/net/adoptium/configuration/JacksonKotlinModule.java [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+
+    @Singleton
+    ObjectMapper objectMapper(Instance<ObjectMapperCustomizer> customizers) {
+        ObjectMapper mapper = new ObjectMapper(); // Custom `ObjectMapper`
+
+        // Apply all ObjectMapperCustomerizer beans (incl. Quarkus)
+        for (ObjectMapperCustomizer customizer : customizers) {
+            customizer.customize(mapper);
+        }
+        System.out.println("Hoi Module");
```
Comment (hendrikebbers) [inline src/main/java/net/adoptium/configuration/JacksonKotlinModule.java:23]: Hoi System.out :D

File: src/main/java/net/adoptium/configuration/JacksonKotlinModule.java [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
+import com.fasterxml.jackson.module.kotlin.KotlinModule;
+import io.quarkus.jackson.ObjectMapperCustomizer;
+
+import javax.enterprise.inject.Instance;
+import javax.inject.Singleton;
+
+
+public class JacksonKotlinModule {
```
Comment (hendrikebbers) [inline src/main/java/net/adoptium/configuration/JacksonKotlinModule.java:13]: What is this class doing? JavaDoc will be important in such class.

File: src/main/java/net/adoptium/configuration/JacksonKotlinModule.java [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+
+    @Singleton
+    ObjectMapper objectMapper(Instance<ObjectMapperCustomizer> customizers) {
```
Comment (hendrikebbers) [inline src/main/java/net/adoptium/configuration/JacksonKotlinModule.java:16]: let's try to avoid methods/classes/fields without explicit access modifier.

**Review by hendrikebbers — COMMENTED** [review]
Some comments

File: pom.xml (hunk 1 above)
Comment (julius-b) [reply pom.xml:56]: jackson-module-kotlin fails to serialize json responses without it, but works when it's included (openjdk-api-v3-models are written in Kotlin)...

**Pull request merged**